# C# 终于

> 原文：<https://www.javatpoint.com/c-sharp-finally>

C# finally 块用于执行重要的代码，无论是否处理异常，都要执行这些代码。它必须以 catch 或 try block 开头。

### C# 最后一个处理异常的例子

```cs
using System;
public class ExExample
{
    public static void Main(string[] args)
    {
        try
        {
            int a = 10;
            int b = 0;
            int x = a / b;
        }
        catch (Exception e) { Console.WriteLine(e); }
        finally { Console.WriteLine("Finally block is executed"); }
        Console.WriteLine("Rest of the code");
    }
}

```

输出:

```cs
System.DivideByZeroException: Attempted to divide by zero.
Finally block is executed
Rest of the code

```

### C# 最后一个不处理异常的例子

```cs
using System;
public class ExExample
{
    public static void Main(string[] args)
    {
        try
        {
            int a = 10;
            int b = 0;
            int x = a / b;
        }
        catch (NullReferenceException e) { Console.WriteLine(e); }
        finally { Console.WriteLine("Finally block is executed"); }
        Console.WriteLine("Rest of the code");
    }
}

```

输出:

```cs
Unhandled Exception: System.DivideBy

```